delivery/client.py

```diff
@@ -416,6 +416,21 @@ def query_metadata_raw(
 
         return res.json()
 
+    def query_metadata(
+        self,
+        components: collections.abc.Iterable[cm.Component]=(),
+        type: dso.model.Datatype | tuple[dso.model.Datatype]=None,
+    ) -> tuple[dso.model.ArtefactMetadata]:
+        artefact_metadata_raw = self.query_metadata_raw(
```

Comment:
nit: if `raw` flavour is not intended to be used, mark it as private (leading underscore)?

Reply:
The `query_metadata` route is just added as a convenience parser, however, parsing the raw data to `dso.model.ArtefactMetadata` comes with some data loss (i.e. patched-in data by the delivery-service). For most cases, this is perfectly fine as those extra information is not required, but sometimes it is and so `query_metadata_raw` has to be instrumented.